API/device/artik053.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2017-present Samsung Electronics Co., Ltd. and other contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base
+
+import connection
+import re
+import signal
+import time
+
+from ..common import (console, paths, utils)
+
+
+class Device(base.DeviceBase):
+    '''
+    Artik053 device class. (Only for testing the testrunner.)
+    '''
+    def __init__(self, options):
+        super(self.__class__, self).__init__('artik053', remote_path='/')
+        self.serial = connection.serialcom.Connection(options, 'TASH>>')
+        self.platform = utils.get_system().lower() + utils.get_architecture()
+
+    def install_dependencies(self):
+        '''
+        Install dependencies of the board.
+        '''
+        pass
+
+    def get_test_path(self):
+        '''
+        Return the test path on the device.
+        '''
+        return '/rom/test'
+
+    def flash(self, os):
+        '''
+        Flash the given operating system to the board.
+        '''
+        utils.execute(paths.TIZENRT_OS_PATH, 'make', ['download', 'ALL'])
+
+    def reset(self):
+        '''
+        Reset the board.
+        '''
+        try:
+            # TODO: Fix error message
+            utils.execute(paths.TIZENRT_OPENOCD_PATH, './%s/openocd' % self.platform, 
+            ['-f', 'artik053.cfg', '-c', 'reset'], quiet=True, errorQuiet=True)
+        except:
+            pass
+
+        # Wait a moment to boot the device.
+        time.sleep(2)
+
+    def login(self):
+        '''
+        Create connection.
+        '''
+        try:
+            self.serial.open()
+
+            # Press enters to start the serial communication and
+            # go to the test folder because some tests require resources.
+            self.serial.exec_command('\n\n')
+            self.serial.exec_command('cd /rom/test')
+
+        except Exception as e:
+            console.fail(str(e))
+
+    def logout(self):
+        '''
+        Close connection.
+        '''
+        self.serial.close()
+
+    def execute(self, cmd, args=[]):
+        '''
+        Run the given command on the board.
+        '''
+        self.reset()
+        self.login()
+        
+        # Send testrunner command to the device and process its result.
+        self.serial.putc('%s %s\n' % (cmd, ' '.join(args).encode('utf8')))
```

Comment:
Could we use `serial.exec_command` in this place? I'm not familiar with TizenRT but it should always give the prompt when iotjs terminates (like NuttX).
Should we check the 'AssertionError', 'uncaughtException', 'arm_dataabort' error messages? I think `regexp` could be a shorter solution to determine the exit value. If the given pattern matches to the stdout, we can read the value easily. If there is no match, maybe an error happened (by an assertion, uncaughtexception or arm_dataabort):

```
stdout = self.serial.exec_command(command)
match = re.search('RESULT: (\d+)', stdout)
if match:
    exitcode = match.group(1)
else:
    exitcode = 1
```

In this case the implementation could be similar to the stm32f4dis target.

Reply:
Unfortunately, when an error occurs, the shell command does not appear.(it must be reset). 
So I have to check the error message from stdout.
And It is also difficult to check the exit value because `echo` command is not supported. I will make improvements every time os updates.
And I updated your opinion to find the exit value using `regexp`.

Reply:
Thanks to clarify that.